crates/cdk/src/mint_url.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2022-2023 Yuki Kishimoto
+// Distributed under the MIT software license
+
+//! Url
+
+use core::fmt;
+use core::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use url::{ParseError, Url};
+
+/// Url Error
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum Error {
+    /// Url error
+    #[error(transparent)]
+    Url(#[from] ParseError),
+}
+
+/// MintUrl Url
+#[derive(Debug, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub struct MintUrl(String);
+
+impl MintUrl {
+    /// New mint url
+    pub fn new<S>(url: S) -> Self
+    where
+        S: Into<String>,
+    {
+        let url: String = url.into();
+        Self(url.trim_end_matches('/').to_string())
+    }
+
+    /// Empty mint url
+    pub fn empty() -> Self {
+        Self(String::new())
+    }
+
+    /// Join onto url
+    pub fn join(&self, path: &str) -> Result<Url, Error> {
+        let url: Url = self.try_into()?;
+        Ok(url.join(path)?)
+    }
+
+    /// Remove trailing slashes from url
+    pub fn trim_trailing_slashes(&self) -> Self {
+        Self(self.to_string().trim_end_matches('/').to_string())
+    }
+}
+
```

Comment:
```suggestion
impl<'de> Deserialize<'de> for MintUrl {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        let s = String::deserialize(deserializer)?;
        MintUrl::from_str(&s).map_err(serde::de::Error::custom)
    }
}
```

I think we should add a custom deserlizer so that if we get a token that does not trim the slash we trim it.